### PR TITLE
Implement LWG-3594 `inout_ptr` - inconsistent `release()` in destructor

### LINKED ISSUE #3218: LWG-3594 `inout_ptr` - inconsistent `release()` in destructor
LWG-3594 `inout_ptr` - inconsistent `release()` in destructor

### PR DESCRIPTION
**Pull request by fsb4000**
Fixes #3218

### CODE REVIEW

**Review by strega-nil-ms — APPROVED** [review]
LGTM, thanks!

**Review by StephanTLavavej — APPROVED** [review]

File: stl/inc/memory [hunk 1]
```diff
@@ -4190,21 +4190,23 @@ public:
     inout_ptr_t(const inout_ptr_t&) = delete;
 
     ~inout_ptr_t() {
+        using _Sp = _Pointer_of_or<_SmartPtr, _Pointer>;
```
[the remark this answers is not in the document]
Comment (fsb4000) [reply stl/inc/memory:4193]: No, I'm just blind 🤦‍♂️ . I thought I wrote `if constexpr (!is_pointer_v<_Sp>)`. 
Yes, you are right, there is no reason to extract `_Sp`.

**Review by CaseyCarter — APPROVED** [review]

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks for improving this feature, and another decrease in the Status Chart's LWG line! :smile_cat: :chart_with_downwards_trend: :sparkles:

**Pull request merged**